### ISSUE TITLE
Feature request: Expose list of changed files in PR in metadata

### ISSUE THREAD

**Issue by annaken** [telia-oss/github-pr-resource#125]
We have a monorepo, so for any given PR I only want to build and run tests relevant to the changed file, rather than the entire repo. I can see that that information is available in the V4 api like
https://developer.github.com/v4/object/pullrequestchangedfile/
but is not currently exposed in the github-pr-resource.

Would it be possible to pass a list of all changes files back to concourse via the metadata?

**Comment by holger-wg2** [comment]
Here is an attempt on implementing the feature: https://github.com/telia-oss/github-pr-resource/pull/128

**Comment by boekkooi-fresh** [comment]
A work around for this would be using something like:
```
git --no-pager diff --name-only --diff-filter=ACMR "$(jq -r '.[] | select (.name == "base_sha").value' .git/resource/metadata.json)"
```

**Comment by itsdalmo** [comment]
This has been implemented in [v0.17.0](https://github.com/telia-oss/github-pr-resource/releases/tag/v0.17.0) by @holger-wg2 and seems to work great 👍 Feel free to reopen if that is not the case
